packages/terra-table/src/subcomponents/Cell.jsx

```diff
@@ -0,0 +1,263 @@
+
+    return focusableElements.length > 0;
+  };
+
+  useEffect(() => {
+    setIsInteractable(hasFocusableElements());
+  }, []);
+
+  /**
+   * Handles the onDeactivate callback for FocusTrap component
+   */
+  const deactivateFocusTrap = () => {
+    setIsFocusTrapEnabled(false);
+    if (gridContext.setCellAriaLiveMessage) {
+      gridContext.setCellAriaLiveMessage(intl.formatMessage({ id: 'Terra.table.resume-navigation' }));
+    }
+  };
+
+  const onMouseDown = ((event) => {
+    if (!isFocusTrapEnabled) {
+      onCellSelect({
+        rowId, columnId, rowIndex, columnIndex, isShiftPressed: event.shiftKey, isCellSelectable: (!isMasked && isSelectable),
+      });
+    }
+  });
+
+  const handleKeyDown = (event) => {
+    const key = event.keyCode;
+
+    if (isFocusTrapEnabled) {
+      switch (key) {
+        case KeyCode.KEY_ESCAPE:
+          deactivateFocusTrap();
+          break;
+        default:
+      }
+      event.stopPropagation();
+    } else {
+      switch (key) {
+        case KeyCode.KEY_RETURN:
+          // Lock focus into component
+          if (hasFocusableElements()) {
+            setIsFocusTrapEnabled(true);
+            if (gridContext.setCellAriaLiveMessage) {
+              gridContext.setCellAriaLiveMessage(intl.formatMessage({ id: 'Terra.table.cell-focus-trapped' }));
+            }
+            event.stopPropagation();
+            event.preventDefault();
+          }
+          break;
+        case KeyCode.KEY_SPACE:
+          if (onCellSelect) {
+            onCellSelect({
+              rowId, columnId, rowIndex, columnIndex, isShiftPressed: event.shiftKey, isCellSelectable: (!isMasked && isSelectable),
+            });
+          }
+          event.preventDefault(); // prevent the default scrolling
+          break;
+        default:
+      }
+    }
+  };
+
+  // Create cell content for masked and blank cells
+  let cellContent;
+  if (isMasked) {
+    cellContent = (
+      <span className={cx('no-data-cell', theme.className)}>
+        {intl.formatMessage({ id: 'Terra.table.maskedCell' })}
+      </span>
+    );
+  } else if (!children) {
+    cellContent = (
+      <span className={cx('no-data-cell', theme.className)}>
+        {intl.formatMessage({ id: 'Terra.table.blank' })}
+      </span>
+    );
+  } else {
+    cellContent = children;
+  }
+
+  const className = cx('cell', {
+    masked: isMasked,
+    pinned: columnIndex < columnContext.pinnedColumnOffsets.length,
+    selectable: isSelectable && !isMasked,
+    selected: isSelected && !isMasked,
+    highlighted: isHighlighted,
+    blank: !children,
+  }, theme.className);
+
+  const cellLeftEdge = (columnIndex < columnContext.pinnedColumnOffsets.length) ? columnContext.pinnedColumnOffsets[columnIndex] : null;
+
+  const CellTag = isRowHeader ? 'th' : 'td';
+
+  return (
+    <CellTag
+      ref={cellRef}
+      aria-selected={isGridContext ? isSelected : undefined}
```

Comment:
We should not have the isGridContext check here.  We can just set it based on the selection state.

Reply:
There is an edge case where `aria-selected={false}` in tables creates an accessibility violation, but since since we're saying that cell selection is a grid functionality (which setting the value is allowed) then we can just honor the value from the cell data. Done.